### DISCUSSION TITLE
No code splitting when including source from monorepo

### DISCUSSION THREAD

**Discussion by Poolshark** [babel/babel#14570] [Q&A] [answered]
I'm trying to figure this out since days now but I simply can't get my head around what's going on. I further don't know if this problem is **babel** related or if it originates from **webpack**.

I have a monorepo with **3 apps** which all rely on components from `packages`
```
frontend/
  apps/       <-- each app is a package in the scope of the Monroe
    app1/
    ...
  packages/   <-- this is a package in the monorepo scope
    components/
    providers/
    ...
```

Inside _app specific_ components I import eg. components like
```tsx
import { MyComponent } from 'packages/components';
```
Inside _packages_ I have aliased imports, eg. requesting a _component_ in _providers_
```tsx
import { MyComponent } from 'frontend/components';
```

The `tsconfig.json` in **packages** defines the local aliases
```
{
  ...
  "paths": {
    "frontend/components": ["./components"],
    ...
}
```

In my `webpack.config.ts` I define the aliases like
```ts
{
  resolve: {
    alias: {
      components: resolve(__dirname, "../../../frontend/apps/campus/src/components") // <-- resembles the absolute path to the config file
    }
  }
}
```

It all works but when I check my bundle I get a **main chunk** with the entire content from **packages**? Does that have something to do with my aliases or should I do something with my chunk settings? 

Here's a screenshot from the bundle analyzer:
![Screenshot 2022-05-20 at 13 18 16](https://user-images.githubusercontent.com/9102922/169517177-c01daa6b-e5f1-476a-b9ff-29ea0e183ed3.png)

Especially interesting is the `iconpack_default`, from which I **dynamically import** icons. I would have expected this to put every icon inside a separate chunk...

I know, this might be quite vague but I'd appreciate any kind of push in the right direction!

Thanks! 🙌

**Comment by nicolo-ribaudo** [comment]
It's more likely that this is caused by webpack rather than Babel, since Babel works on a per-file basis.

The only Babel-related suggestion I have is to make sure that you aren't compiling dynamic imports: if you are using `preset-env`, pass to it the `debug: true` option and see which plugins are enabled.

**Reply by Poolshark** [reply]
Hi @nicolo-ribaudo!

Thanks for the quick answer!


How would I see that I don't compile dynamic imports? Is this `syntax-dynamic-imports` which appears under `Using plugins` in the debug output?

By the way, this is my `optimization` setting:

```
minimize: true,
moduleIds: "deterministic",
nodeEnv: "production",
splitChunks: {
  chunks: "all",
  enforceSizeThreshold: 2000,
},
```

Is there something I should do differently?

**Reply by nicolo-ribaudo** [reply]
Ok, if you see `syntax-dynamic-imports` it means that your Babel setup it's correct and it's not transforming dynamic imports (`syntax` means "just parse").

I suggest asking in the webpack support spaces, since here you are unlikely to find a webpack expert :sweat_smile:

**Reply by Poolshark** [reply]
Ok, thank you!

Setting the `debug` option and babel presets to
```
presets: [
            [
              "@babel/preset-env",
              {
                useBuiltIns: "entry",
                debug: true,
                exclude: ["proposal-dynamic-import"],
                modules: false,
                corejs: 3,
                // targets: "> 0.25%, not dead",
              },
            ],
            ["@babel/preset-react", { runtime: "automatic" }],
            "@babel/preset-typescript",
```
now also shows me
```
The entry point for the corejs3 polyfill has not been found.
```
basically for all files in the **packages** package. Could that be a related issue?

**Reply by nicolo-ribaudo** [reply]
Nope, it's just telling you that for most files it's not I including polyfills (because they are included in the file where you import core-js).

**Comment by Poolshark** [comment, marked as answer]
The answer is not `babel` related and can be resolved by using `sideEffects` properly.

See full answer [@webpack](https://github.com/webpack/webpack/discussions/15854) or [@stackoverflow](https://stackoverflow.com/questions/72352675/re-exports-not-properly-bundled-with-webpack-5/72372546#72372546).
